marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/endpoints/collector.sjs

```diff
@@ -54,16 +54,16 @@ if(method === 'GET') {
     resp = fn.error(null, "RESTAPI-SRVEXERR", Sequence.from([404, "Not Found", "The requested flow was not found"]));
   } else {
     let stepDoc = flowDoc.steps[step];
-    let baseStep = datahub.flow.step.getStepByNameAndType(stepDoc.name, stepDoc.type);
+    let baseStep = datahub.flow.step.getStepByNameAndType("default-ingest", "INGEST");
```

Comment:
Why is the flow name and type hardcoded ? Also for INGEST flows, collector is not run

Reply:
@srinathgit i tried to understand and fix the exception. I guess the baseStep is our default. The stepDoc is the customized step. if I pass flow name (flow1 for example) and type (INGEST), datahub.flow.step.getStepByNameAndType return null.  I am not quite sure how to fix it correctly. But I works after the change. I might miss something here.

Reply:
@srinathgit Query step runner works for you?